packages/common/src/converter/payload-converter.ts

```diff
@@ -319,6 +333,60 @@ export class SearchAttributePayloadConverter implements PayloadConverter {
 
 export const searchAttributePayloadConverter = new SearchAttributePayloadConverter();
 
+export class TypedSearchAttributePayloadConverter implements PayloadConverter {
+  jsonConverter = new JsonPayloadConverter();
+
+  public toPayload<T>(typedSearchAttribute: T): Payload {
+    if (!isTypedSearchAttribute(typedSearchAttribute)) {
+      throw new ValueError('Invalid typed search attribute');
+    }
+
+    const type = typedSearchAttribute[0];
+    let value = typedSearchAttribute[1];
+    if (type === 'DATETIME') {
+      // Convert Date to ISO string
+      value = (value as Date).toISOString();
+    }
+
+    const payload = this.jsonConverter.toPayload(value);
+    if (payload === undefined) {
+      throw new ValueError('Could not convert typed search attribute to payload');
+    }
+    // Note: this shouldn't be the case but the compiler complains without this check.
+    if (payload.metadata === undefined || payload.metadata === null) {
```

Comment:
Please you the following idiom whenever you get this:

```suggestion
    if (payload.metadata == null) {
```

For reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/null#difference_between_null_and_undefined

Reply:
Done.